### DISCUSSION TITLE
Missing qemu-system runner in musl-libc targets

### DISCUSSION THREAD

**Discussion by nistvan86** [cross-rs/cross#1107] [Q&A]
I've noticed that `armv7-unknown-linux-musleabihf` doesn't contain qemu-system runner while `armv7-unknown-linux-gnueabihf` does. Checking [targets.toml](https://github.com/cross-rs/cross/blob/main/targets.toml) it's also omitted for other architectures with this C library. Is there a particular reason for that?
Since qemu is built from source in your Docker image I cannot simply use the Ubuntu apt package manager to add it afterwards using the custom Dockerfile option.

**Comment by Alexhuszagh** [comment]
None of our musl targets (and even some of our glibc targets, such as `arm-unknown-linux-gnueabi`, `arm-unknown-linux-gnueabihf`, `i586-unknown-linux-gnu`) currently have Linux images. @Emilgardis we could possibly implement this, couldn't we? We control the Linux versions for the crosstool builds, and can also control it in our [musl builds](https://github.com/richfelker/musl-cross-make/blob/fe915821b652a7fa37b34a596f47d8e20bc72338/Makefile#L11). 

The only issue is I believe our Linux image cannot be newer than the headers we compile against? This likely won't be an issue in practice, but in rare, rare, rare cases it could be. Currently, we build musl against Linux headers for `4.19.88-1` and crosstool against Linux headers for `4.19.21`, while the current Debian system packages either use a 5.x or 6.x version except for 32-bit BE MIPS. That being said, I doubt it will be an issue since Linux tries not to break userspace and we'd solely be unable to use newer Linux features.


For `i586-unknown-linux-gnu`, we can't do it because the `i386` port of Debian requires `i686` instructions, so the Linux image won't actually be compatible with a native `i586` system.
